classification1.Rmd

```diff
@@ -472,86 +489,155 @@ perim_concav + annotate("path",
+y2 <- c(pull(new_obs_3[2]), data$Concavity[2])
+z2 <- c(pull(new_obs_3[3]), data$Symmetry[2])
+
+x3 <- c(pull(new_obs_3[1]), data$Perimeter[3])
+y3 <- c(pull(new_obs_3[2]), data$Concavity[3])
+z3 <- c(pull(new_obs_3[3]), data$Symmetry[3])
+
+x4 <- c(pull(new_obs_3[1]), data$Perimeter[4])
+y4 <- c(pull(new_obs_3[2]), data$Concavity[4])
+z4 <- c(pull(new_obs_3[3]), data$Symmetry[4])
+
+x5 <- c(pull(new_obs_3[1]), data$Perimeter[5])
+y5 <- c(pull(new_obs_3[2]), data$Concavity[5])
+z5 <- c(pull(new_obs_3[3]), data$Symmetry[5])
+
+plot_3d  %>%
+  add_trace(x = x1, y = y1, z = z1, type = "scatter3d", mode = "lines", 
+            name = "lines", showlegend = FALSE, color = I("steelblue2")) %>%
+  add_trace(x = x2, y = y2, z = z2, type = "scatter3d", mode = "lines", 
+            name = "lines", showlegend = FALSE, color =  I("steelblue2")) %>%
+  add_trace(x = x3, y = y3, z = z3, type = "scatter3d", mode = "lines", 
+            name = "lines", showlegend = FALSE, color =  I("steelblue2")) %>%
+  add_trace(x = x4, y = y4, z = z4, type = "scatter3d", mode = "lines", 
+            name = "lines", showlegend = FALSE, color =  I("orange2")) %>%
+  add_trace(x = x5, y = y5, z = z5, type = "scatter3d", mode = "lines", 
+            name = "lines", showlegend = FALSE, color =  I("steelblue2")) 
 ```
 
 *Click and drag the plot above to rotate it, and scroll to zoom. Note that in
 general we recommend against using 3D visualizations; here we show the data in
-3D only to illustrate what "higher dimensions" and "nearest neighbours" look like,
+3D only to illustrate what "higher dimensions" and "nearest neighbors" look like,
 for learning purposes.*
 
-### Summary of K-nearest neighbours algorithm
 
-In order to classify a new observation using a K-nearest neighbour classifier, we have to:
+### Summary of $K$-nearest neighbors algorithm
+
+In order to classify a new observation using a $K$-nearest neighbor classifier, we have to:
 
 1. Compute the distance between the new observation and each observation in the training set.
 2. Sort the data table in ascending order according to the distances.
 3. Choose the top $K$ rows of the sorted table.
-4. Classify the new observation based on a majority vote of the neighbour classes.
+4. Classify the new observation based on a majority vote of the neighbor classes.
 
 
-## K-nearest neighbours with `tidymodels`
+## $K$-nearest neighbors with `tidymodels`
 
-Coding the K-nearest neighbours algorithm in R ourselves can get complicated,
+Coding the $K$-nearest neighbors algorithm in R ourselves can get complicated,
 especially if we want to handle multiple classes, more than two variables,
 or predicting the class for multiple new observations. Thankfully, in R,
-the K-nearest neighbours algorithm is implemented in the `parsnip` package 
+the $K$-nearest neighbors algorithm is implemented in the `parsnip` package 
 included in the 
 [`tidymodels` meta package](https://www.tidymodels.org/), along with 
```

Comment:
do we use the term "meta package' anywhere else?

in fact I think when we introduce tidyverse, we specifically just call it a package

Reply:
I searched and we use "metapackage" or "meta package" in:
- jupyter.Rmd
- viz.Rmd
- wrangling.Rmd
Should we remove all instances of it? If so, I will make another issue.

Reply:
@leem44 no -- if we use it in a bunch of places it's fine to just leave it as-is (for this PR, anyway)